scripts/actions/utils/vendor-request.js

```diff
@@ -1,19 +1,53 @@
+'use strict';
+
 const fetch = require('node-fetch');
 
-/** @throws {Error} Will throw an error if the response "code" is not 'SUCCESS' */
-const makeRequest = async (url, options) => {
-  const resp = await fetch(url.href, options);
-  const { response } = await resp.json();
-  const { code, data } = response;
+const MAX_RETRY = 5;
+const DELAY_TIME = 1500;
+// let nthTry = 1;
+const waitFor = (millSeconds) => {
+  return new Promise((resolve) => {
+    setTimeout(() => {
+      resolve();
+    }, millSeconds);
+  });
+};
 
-  if (code !== 'SUCCESS') {
-    console.error(
-      `[!] Unable to make a ${options.method} request to ${url.href}. (${code})`
+/**
+ * @param {Object} options
+ * @param {String} url
+ * @param {Number} nthTry
+ * @returns {Object} data The result after making the request.
+ * @throws {Error} Will throw an error if the response "code" is not 'SUCCESS' after retrying
+ */
+const makeRequest = async (url, options, nthTry) => {
+  nthTry = nthTry || 1;
+  try {
+    const resp = await fetch(url.href, options);
+    const { response } = await resp.json();
+    const { code, data } = response;
+    if (code !== 'SUCCESS') {
+      throw new Error(JSON.stringify(response, null, 2));
+    }
+    console.log(`Successful response`);
+    return data;
+  } catch (e) {
+    if (nthTry === MAX_RETRY) {
+      console.error(
+        `[!] Unable to make a ${options.method} request to ${url.href} after ${MAX_RETRY} attempts.`
+      );
+      return Promise.reject(e);
+    }
+    ++nthTry;
+    console.warn(
+      `[!] Error making request on attempt ${nthTry}/${MAX_RETRY}. Retrying in ${
+        DELAY_TIME / 1000
+      } seconds`
     );
-    throw new Error(JSON.stringify(response, null, 2));
+    // wait for delayTime amount of time before calling this method again
+    await waitFor(DELAY_TIME);
+    return makeRequest(url, options, nthTry);
   }
```

Comment:
i will write some thoughts since i wanted to review this, but since my javascript knowledge is not the strongest, i think it would be worth grabbing another set of eyes 🙈 

1. im not a fan of the use of recursion here. i think a loop would be easier to read & understand whats happening.
2. it looks like youre using a try/catch as an if/else based on the fetch output, and that seems a bit awkward to me.
3. for the last line `return makeRequest` does that not need an await since the method is async?

Reply:
im also still in favor of using a library, since hopefully someone has already worked out some of this complexity & has some of the features i mentioned in an overall pr comment, if you happen to find a good candidate. feel free to veto this though.

Reply:
+1 to Kris's comment. Using a library for this logic reduces the amount of code and tests we have to maintain. @nr-kkenney found a library, https://github.com/mauricedb/polly-js, that could be perfect for this.

Reply:
I'd love to grab some time in a campfire to chat about it, cause I actually really like the use of recursion here. I'd probably restructure things a little to reduce the logic, but this makes sense to my brain:

```js
const makeRequest = async (url, options, retriesRemaining = MAX_RETRY) => {
  try {
    // make an async call
  } catch (e) {
    if (retriesRemaining === 0) {
      // bail out
    } else {
      await sleep(DELAY_TIME);
      return makeRequest(url, options, retriesRemaining - 1);
    }
  };
}
```